### ISSUE TITLE
"downloaded archive of binary target 'grpc' does not contain a binary artifact"

### ISSUE THREAD

**Issue by lukehutch** [grpc/grpc-ios#256]
When I try to build a Flutter project in Xcode that uses Google APIs, the build fails with this error:

```
Running Xcode build...                                                  
Xcode archive done.                                         50.1s
Failed to build iOS app
Uncategorized (Xcode): Could not resolve package dependencies:
  downloaded archive of binary target 'grpc' from 'https://dl.google.com/firebase/ios/bin/grpc/1.62.2/rc0/grpc.zip' does not contain a
  binary artifact.
  fatalError

Encountered error while archiving for device.
```

I have tried many combinations of deleting `Pod.lock`, `Pods`, `DerivedData`, etc.; `flutter clean`; cleaning the build in Xcode, etc., but I can't seem to fix this.

I downloaded the zipfile, and it exists and contains binaries.

**Comment by lukehutch** [comment]
`Package.resolved` contains:

```
    {
      "identity" : "grpc-binary",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/google/grpc-binary.git",
      "state" : {
        "revision" : "e9fad491d0673bdda7063a0341fb6b47a30c5359",
        "version" : "1.62.2"
      }
    },
```

and it does not contain any reference to a package `grpc`.

**Comment by albermoon** [comment]
do you find any solution? I have a similar problem

**Comment by lukehutch** [comment]
Yes, the only thing that worked (after many days of searching online and trying a million things) was to move the `ios` directory to `ios_old`, then use `flutter create` to create a new blank ios directory, then manually copy back in the things that I had manually configured in the first project. Primarily I think what changed is that there was less Cocoapods integration in the recreated iOS project.

**Comment by danilocatalanoapsystem** [comment]
I have this problem. There is a solution?

**Comment by elcentaurx** [comment]
> do you find any solution? I have a similar problem

did you find any solution?

**Comment by jaymeschroeder** [comment]
I had this problem yesterday, and saw this thread with no answers during my search. There is no answer to this anywhere online so I figured I'd come and add my answer which solved it for me.

When building a flutter app in xcode, all dependencies are pulled from the pubspec.yaml file. The problem I ran into was that I had mistakenly added the firebase-ios-sdk dependency within Xcode itself, and this was causing my project to try to build the app with discrepancies about where to download firebase. Removing the firebase-ios-sdk from xcode solved the issue.

**Comment by aslam9691** [comment]
This issue appears to stem from a dependency management conflict between Swift Package Manager and CocoaPods within Flutter projects. Can anyone reproduce this error in a minimal, pure native iOS project to help isolate the cause and investigate further?
